### ISSUE TITLE
Integration test suite failing 

### ISSUE THREAD

**Issue by rajaskakodkar** [kubernetes/kubernetes#92700]
I am facing test case failures while running integration test cases using test-dockerized.sh with an added flag "-p 1" to disable parallelism. Following failures are observed on 1.18.4 branch

```
=== Failed
=== FAIL: test/integration/apiserver/apply  (0.00s)
I0629 02:44:47.759999   23295 etcd.go:81] etcd already running at http://127.0.0.1:2379
FAIL	k8s.io/kubernetes/test/integration/apiserver/apply	106.723s

=== FAIL: test/integration/auth  (0.00s)
I0629 02:47:31.644814   23520 etcd.go:81] etcd already running at http://127.0.0.1:2379
FAIL	k8s.io/kubernetes/test/integration/auth	73.154s

=== FAIL: test/integration/daemonset  (0.00s)
I0629 02:53:17.668164   23978 etcd.go:81] etcd already running at http://127.0.0.1:2379
FAIL	k8s.io/kubernetes/test/integration/daemonset	300.637s

=== FAIL: test/integration/master TestReconcilerMasterLeaseMultiCombined (20.49s)

=== FAIL: test/integration/replicaset  (0.00s)
I0629 03:13:51.149436   25639 etcd.go:81] etcd already running at http://127.0.0.1:2379
FAIL	k8s.io/kubernetes/test/integration/replicaset	77.660s

=== FAIL: test/integration/scheduler  (0.00s)
I0629 03:16:34.260033   25865 etcd.go:81] etcd already running at http://127.0.0.1:2379
FAIL	k8s.io/kubernetes/test/integration/scheduler	84.373s

=== FAIL: test/integration/volume  (0.00s)
I0629 03:20:53.646581   26343 etcd.go:81] etcd already running at http://127.0.0.1:2379
FAIL	k8s.io/kubernetes/test/integration/volume	122.375s
```
However, if I run each of these modules individually, I see that only one of the test cases fails from each module
with the error
```
I0701 03:46:24.334858    2750 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{http://127.0.0.1:2379  <nil> 0 <nil>}]
I0701 03:46:43.342529    2750 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
E0701 03:46:43.342551    2750 master_utils.go:197] error in bringing up the master: error building core storage: context deadline exceeded
W0701 03:46:43.342618    2750 clientconn.go:1208] grpc: addrConn.createTransport failed to connect to {http://127.0.0.1:2379  <nil> 0 <nil>}: didn't receive server preface in time. Reconnecting...
F0701 03:46:43.342680    2750 master_utils.go:199] error in bringing up the master: error building core storage: context deadline exceeded
```
and if that test case is run individually, it passes

How can I get the test suite to pass? Following are my specs
```
Specs:
Arch: x86_64
OS: Ubuntu 18.04
CPUs: 4
go version: go version go1.14.4 linux/amd64
```
Attaching log file for reference
[INTEL1.18-p12.log](https://github.com/kubernetes/kubernetes/files/4858018/INTEL1.18-p12.log)

**Comment by neolit123** [comment]
> sig api-machinery

based:
> etcd already running at http://127.0.0.1:2379

but feel free to adjust.

**Comment by hasheddan** [comment]
/milestone v1.19
/cc @kubernetes/ci-signal 
/priority critical-urgent

@roycaihw @jingyih any updates here? I am now seeing this failure on https://testgrid.k8s.io/sig-release-master-blocking#integration-master

**Comment by RobertKielty** [comment]
From the last failure we have (has been failing for the last 32 runs) 

[sig-release-master-blocking#integration-master](https://testgrid.k8s.io/sig-release-master-blocking#integration-master)

we have an from the following Prow run [1282672402110615552](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-integration-master/1282672402110615552)

> === Skipped
> === SKIP: test/integration/apiserver/apply TestApplyStatus/apiregistration.k8s.io/v1,_Resource=apiservices (0.00s)
>     TestApplyStatus/apiregistration.k8s.io/v1,_Resource=apiservices: status_test.go:143: 
>     --- SKIP: TestApplyStatus/apiregistration.k8s.io/v1,_Resource=apiservices (0.00s)
> 
> === SKIP: test/integration/apiserver/apply TestApplyStatus/apiregistration.k8s.io/v1beta1,_Resource=apiservices (0.00s)
>     TestApplyStatus/apiregistration.k8s.io/v1beta1,_Resource=apiservices: status_test.go:143: 
>     --- SKIP: TestApplyStatus/apiregistration.k8s.io/v1beta1,_Resource=apiservices (0.00s)
> 
> === SKIP: test/integration/client TestMultiWatch (0.00s)
>     TestMultiWatch: client_test.go:535: 
> 
> === SKIP: test/integration/master TestUpdateNodeObjects (0.00s)
>     TestUpdateNodeObjects: synthetic_master_test.go:722: UPDATE_NODE_APISERVER is not set
> 
> 
> === Failed
> === FAIL: test/integration/ipamperf  (0.00s)
> I0713 14:08:29.954589  111206 etcd.go:81] etcd already running at http://127.0.0.1:2379
> FAIL	k8s.io/kubernetes/test/integration/ipamperf	600.254s
> 
> === FAIL: test/integration/scheduler_perf  (0.00s)
> I0713 14:10:56.190491  112593 etcd.go:81] etcd already running at http://127.0.0.1:2379
> FAIL	k8s.io/kubernetes/test/integration/scheduler_perf	600.415s
> 
> 
> DONE 2842 tests, 4 skipped, 2 failures in 13.272s

Full logs here [ci-kubernetes-integration-master/1282672402110615552/build-log.txt](https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-integration-master/1282672402110615552/build-log.txt)

**Comment by roycaihw** [comment]
@hasheddan @RobertKielty It seems that the failure of https://testgrid.k8s.io/sig-release-master-blocking#integration-master was caused by https://github.com/kubernetes/test-infra/pull/18196#issuecomment-657975786. It looks like that PR added some tests to the prowjob, which aren't supposed to be run in parallel, and therefore hit the "etcd already running" error. 

I think the `sig-release-master-blocking#integration-master` failure is unrelated to this issue. Since the failure was related to prowjob config change introduced two days ago, and this issue was about running integration tests manually

**Comment by RobertKielty** [comment]
Thank you @roycaihw for triaging this.

**Comment by roycaihw** [comment]
Please see my comment in https://github.com/kubernetes/test-infra/pull/18196#issuecomment-658000292. The message "etcd already running at http://127.0.0.1:2379" is not an error. For `sig-release-master-blocking#integration-master` it looks like those new tests were timing-out (~600s). For this issue there can be different causes. I will look into the logs.

**Comment by roycaihw** [comment]
@rajaskakodkar Unfortunately the provided log file didn't capture much useful info. E.g. 
```
$ grep -i "Apply" ~/Downloads/INTEL1.18-p12.log
✖  test/integration/apiserver/apply (1m46.723s)
=== FAIL: test/integration/apiserver/apply  (0.00s)
FAIL	k8s.io/kubernetes/test/integration/apiserver/apply	106.723s
```

I don't think the failure is related to running in parallel or not, but we need more verbose log to know what were the actual errors. Could you try reproducing the failure with [higher verbose level](https://github.com/kubernetes/kubernetes/blob/c46c1c043ee8de4c021c03ddb0fb157d1c1ba3f8/hack/make-rules/test-integration.sh#L35) (I'm not sure if this is the only place that controls how log/errors are streamed. You may need to try and see what value is enough) and send us the log?

**Comment by rajaskakodkar** [comment]
@roycaihw I just tried running the following and the test cases passed successfully.
```
sed -i "s/--timeout=120/--timeout=300/" $GOPATH/src/k8s.io/kubernetes/hack/make-rules/test.sh
make test-cmd
make test-integration KUBE_TEST_ARGS="-p 1" 
```
I guess this issue can be closed.

**Comment by roycaihw** [comment]
@rajaskakodkar Thanks! So this could also be timeout-related

/close

**Comment by RobertKielty** [comment]
This job is still failing https://testgrid.k8s.io/sig-release-master-blocking#integration-master

**Comment by RobertKielty** [comment]
> @roycaihw I just tried running the following and the test cases passed successfully.
> 
> ```
> sed -i "s/--timeout=120/--timeout=300/" $GOPATH/src/k8s.io/kubernetes/hack/make-rules/test.sh
> make test-cmd
> make test-integration KUBE_TEST_ARGS="-p 1" 
> ```
> 
> I guess this issue can be closed.

It can be closed when a PR is submitted to fix the job in CI. Until then it will stay open.

**Comment by hasheddan** [comment]
@RobertKielty this issue is not reflecting the failure we are seeing on https://testgrid.k8s.io/sig-release-master-blocking#integration-master. Please see https://github.com/kubernetes/test-infra/pull/18196#issuecomment-658000292. It may be good to open another issue to track that. I plan on submitting a fix today.

/close
